### DISCUSSION TITLE
cant flash FerraniteCrafts 512 kb card

### DISCUSSION THREAD

**Discussion by mer2329** [sanni/cartreader#974] [Ideas & Suggestions]
im  not sure if its me doing somthing wrong, or if its not supported.
if its not supported, i would like to request it to get supported.
if it is supported please tell me which setting to use. i just updated the OSCR to the latest.
heres the product page for it https://ferrantecrafts.com/products/flash-cartridge-for-game-boy-512kb

**Comment by mer2329** [comment]
if it helps, i can post pics of the PCB and closeups of the chips.

**Comment by sanni** [comment]
So far only 29F033, 29F032, 29F016, 29F080 flash chips are supported. In addition to those there is the CFI option that will support some more flash chips if they support the CFI protocol.

Does the big chip say something like 29F040 on it?

**Comment by mer2329** [comment]
the big chip has a part number of SST39SF020A-70-4C-NH. based on my interpetation of the datasheet thats the flash chip. not sure where in the data sheet it specifies protocols supported

that part number keeps sending me to a microchip datasheet, but the chip dosent have a microchip logo on it. so heres a BIG pic the chip you asked about
![IMG1201_045](https://github.com/sanni/cartreader/assets/6768243/a54578e4-d2a5-4ce4-bd69-76bc6418fec1)

**Comment by sanni** [comment]
Replace GB.ino in the current 13.5 Release with GB.ino from this post, upload, and try how far it works when choosing the new "39SF Cart (Audio)" option

**Comment by mer2329** [comment]
ok, its compiled and trying to run. but ive been stuck at writting flash with no progress tried both 5 and 3 volt. been sitting here for a few miniutes
![feem_d75828324e233_IMG_20240707_164606393_HDR](https://github.com/sanni/cartreader/assets/6768243/f7ae39bc-87f0-47d6-ab7c-dba824b5615c)
edit: it did erase successfully though

**Comment by mer2329** [comment]
i will have to do this later as i just had to reinstall windows due to somthing unrelated to this.

**Comment by mer2329** [comment]
ok, i tested and its still failing. no progress on the bar. i would log via serial, but when i enable the serial monitor compinlation fails.

whats the reason that the UI shows MBC5 (Audio)

**Comment by mer2329** [comment]
bump

**Comment by sanni** [comment]
I made some changes today maybe it works now.
https://github.com/sanni/cartreader/archive/refs/heads/master.zip

**Comment by mer2329** [comment]
thanks. i flashed it to my reader and it still is failing to write. if your in the USA i could send you my cart for yourself.

**Comment by sanni** [comment]
One more thing to try is to replace the 5 here for a 3
https://github.com/sanni/cartreader/blob/master/Cart_Reader/GB.ino#L262
Maybe it works better with the MBC3 flash code.

**Comment by mer2329** [comment]
i tried but that didnt work either. i might be doing somthing wrong. 
my offer to mail you the card stands as long as as its in north america. if you are willing to accept. if you accept it, i dont expect to recieve it back even if it dosent get implimented.
